src/content/docs/style-guide/writing-docs/article-templates/branching-docs-guide.mdx

```diff
@@ -0,0 +1,254 @@
+* agentConfig 
+    * this is for the APM agent yml config file
+* appInfo 
+    * this step is the one where you display the content (includes recommendedGuided content)
+
+## A sample MDX file [#sample-mdx]
+
+```
+--- 
+ headingText: Let's begin
+ componentType: default
+ ---
+
+ Boil your water. 
+```
+
+## The appInfo MDX file [#sample-appinfo]
+
+```
+---
+ componentType: appInfoConfig
+ headingText: Tell us about your java app
+ descriptionText: Select how your app is deployed and your framework. The page updates with the instructions you need for your exact Java setup, so you can start monitoring your app’s performance.
+ ---
+
+ ### We recommend the guided install for your app
+
+ A single CLI-command you can run on your system to start monitoring your Java app.
+ Java apps deployed via Tomcat, Wildfly/Jboss, and Jetty (standalone).
+
+ <Button variant="primary">Guided Install</Button>
+
+ If you would prefer to install manually you can follow the steps below.
+```
+
+## The agentConfig MDX file [#sample-agentconfig]
+
+The `agentConfig` MDX file points to the appropriate config file in the install directory and controls what lines in the config file are editable.
+
+```
+ ---
+ componentType: agentConfig
+ headingText: Choose your download option
+ descriptionText: Set your app name and license key. Then download your custom config file. You can also use more advanced config setting, see our docs.
+ inputOptions: 
+   - label: 'Name your app'
+     name: "My app name"
+     codeLine: 29
+     defaultValue: 'My Application'
+     toolTip: 'some random info text' 
+   - label: 'Enter your '
+     url: 
+       href: 'http://one.newrelic.com/api-keys'
+       title: 'New Relic license key'
+     name: licenseKey
+     codeLine: 15
+     defaultValue: '12345'  
+ configFilePath: 'assets/newrelic.yml'
+ ---
+
+ When using the config file, we recommend you: 
+   * change the default `newrelic.yml` permissions to read/write only for the owner of the application server process.
+   * make sure `newrelic.yml` is part of your backup routine.
+   * use the [New Relic Diagnostics CLI](https://docs.newrelic.com/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/run-diagnostics-cli-nrdiag/) to validate your settings, either before or after you deploy.
+```
+
+## MDX frontmatter fields [#mdx-frontmatter]
+
+* headingText: This is the title of the step (required)
+* componentType: (use one of these four options) if missing, defaults to “default” type
+    * default: free mdx file
+    * appInfoConfigOption: this is for dropdowns, use whenever you want dropdowns
+    * agentConfig: this if for the yaml file
+    * appInfoConfig: this step is the one where you display the content (includes recommendedGuided content)
+* descriptionText: The first text in the step content (smaller grey text that acts as a second heading)
+* optionType: only added for appInfoConfigOption
+
+
+whatsNextFilePath: needs to be in your current directory
```

Comment:
@paperclypse is this a front matter field? If so, why isn't it in the list?

Reply:
It's not a frontmatter field. It's just a special, required MDX filetype that I wasn't sure where to put in this doc. I'll get it sorted.